File: projects/ruoshan-law/java-backend/src/main/java/com/ruoshan/entity/Lawyer.java
package com.ruoshan.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "lawyers")
@Data
public class Lawyer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String title;
    
    @Column(length = 1000)
    private String introduction;
    
    private String photoUrl;
    
    private String specialty;
    
    private String email;
    
    private String phone;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: projects/ruoshan-law/vue-frontend/src/api/lawyer.ts
import apiClient from './client'

export interface Lawyer {
  id: number
  name: string
  title: string
  introduction: string
  photoUrl: string
  specialty: string
  email: string
  phone: string
}

export const lawyerApi = {
  getAll: () => apiClient.get<Lawyer[]>('/lawyers'),
  getById: (id: number) => apiClient.get<Lawyer>(`/lawyers/${id}`),
  getBySpecialty: (specialty: string) => apiClient.get<Lawyer[]>(`/lawyers/search?specialty=${specialty}`)
}
